packages/forms/src/patterns/social-security-number/social-security-number.ts

```diff
@@ -0,0 +1,101 @@
+import * as z from 'zod';
+
+import { type SocialSecurityNumberProps } from '../../components.js';
+import {
+  type Pattern,
+  type PatternConfig,
+  validatePattern,
+} from '../../pattern.js';
+import { getFormSessionValue } from '../../session.js';
+import {
+  safeZodParseFormErrors,
+  safeZodParseToFormError,
+} from '../../util/zod.js';
+
+const configSchema = z.object({
+  label: z.string().min(1),
+  required: z.boolean(),
+  hint: z.string().optional(),
+});
+
+export type SocialSecurityNumberPattern = Pattern<z.infer<typeof configSchema>>;
+
+export type SocialSecurityNumberPatternOutput = z.infer<
+  ReturnType<typeof createSSNSchema>
+>;
+
+export const createSSNSchema = (data: SocialSecurityNumberPattern['data']) => {
+  const ssnSchema = z.string().refine(value => {
+    const isValidFormat = /^\d{3}-\d{2}-\d{4}$|^\d{9}$/.test(value);
+    const stripped = value.replace(/[^0-9]/g, '');
+    return isValidFormat && stripped.length === 9;
+  }, 'Social Security Number must contain exactly 9 digits and be formatted as XXX-XX-XXXX or XXXXXXXXX');
```

Comment:
This is a good start for validation. There's some additional guidance on validation rules that USWDS recommends checking for. Here is the link to the validation rules that the SSA provides. https://www.ssa.gov/kc/SSAFactSheet--IssuingSSNs.pdf. I think it would be good to capture these in a `.refine` statement that's chained on to the rule you have here.

Reply:
Thank you for sharing the pdf! I will definitely update the validation hook.

Reply:
Cool. Thank you. I'll review the rest of the changes. I just gave it a quick look earlier today when I left my previous comment.